vm/actor/src/builtin/power/mod.rs

```diff
@@ -43,15 +58,692 @@ impl Method {
+                &TokenAmount::new(0),
+            )
+            .map_err(|e| {
+                ActorError::new(
+                    ExitCode::ErrIllegalState,
+                    format!("failed to subtract pledge for reward: {}", e),
+                )
+            })
+        })?;
+
+        rt.send(
+            &reporter,
+            MethodNum(METHOD_SEND as u64),
+            &Serialized::default(),
+            &reward,
+        )?;
+
+        Self::delete_miner_actor(rt, &fault.target)
+    }
+    pub fn on_epoch_tick_end<BS, RT>(rt: &RT) -> Result<(), ActorError>
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        rt.validate_immediate_caller_is(std::iter::once(&*CRON_ACTOR_ADDR));
+
+        let rt_epoch = rt.curr_epoch();
+        let cron_events = rt
+            .transaction::<_, Result<_, String>, _>(|st: &mut State| {
+                let mut events = Vec::new();
+                for i in st.last_epoch_tick.0..=rt_epoch.0 {
+                    // Load epoch cron events
+                    let epoch_events = st.load_cron_events(rt.store(), ChainEpoch(i))?;
+
+                    // Add all events to vector
+                    events.extend_from_slice(&epoch_events);
+
+                    // Clear loaded events
+                    if !epoch_events.is_empty() {
+                        st.clear_cron_events(rt.store(), ChainEpoch(i))?;
+                    }
+                }
+                st.last_epoch_tick = rt_epoch;
+                Ok(events)
+            })
+            .map_err(|e| {
+                ActorError::new(
+                    ExitCode::ErrIllegalState,
+                    format!("Failed to clear cron events: {}", e),
+                )
+            })?;
+
+        for event in cron_events {
+            // TODO switch 12 to OnDeferredCronEvent on miner actor impl
+            rt.send(
+                &event.miner_addr,
+                MethodNum(12),
+                &event.callback_payload,
+                &TokenAmount::new(0),
+            )?;
+        }
+
+        Ok(())
+    }
+
+    fn delete_miner_actor<BS, RT>(rt: &RT, miner: &Address) -> Result<(), ActorError>
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        let amount_slashed: TokenAmount = rt
+            .transaction::<_, Result<_, String>, _>(|st: &mut State| {
+                st.delete_claim(rt.store(), miner)?;
+
+                st.miner_count -= 1;
+
+                if st.has_detected_fault(rt.store(), miner)? {
+                    st.delete_detected_fault(rt.store(), miner)?;
+                }
+
+                let mut table = BalanceTable::from_root(rt.store(), &st.escrow_table)?;
+                let balance = table.remove(miner)?;
+
+                st.escrow_table = table.root()?;
+
+                Ok(balance)
+            })
+            .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e))?;
+
+        // TODO switch 6 to OnDeleteMiner on miner actor impl
```

Comment:
Can't you do this now?

Reply:
I want to keep changes contained, those method definitions don't exist in the miner actor (it's empty rn)